### PR TITLE
Add a command for searching groupchats

### PR DESCRIPTION
**Pull request by mia-pi-git**
See https://www.smogon.com/forums/threads/3514540/page-18#post-8709906.

### CODE REVIEW

File: server/chat-plugins/chatlog.ts [hunk 1]
```diff
@@ -1333,4 +1333,29 @@ export const commands: ChatCommands = {
 		const [type, userid] = target.split(',').map(toID);
 		return this.parse(`/j view-logsaccess-${type || 'all'}${userid ? `-${userid}` : ''}`);
 	},
+	gcsearch: 'groupchatsearch',
+	async groupchatsearch(target, room, user) {
+		this.checkCan('lock');
+		// eslint-disable-next-line no-useless-escape
```
Comment (Zarel) [inline server/chat-plugins/chatlog.ts:1339]: ?

Comment (mia-pi-git) [reply server/chat-plugins/chatlog.ts:1339]: For some reason, ESLint sees the \- as a useless escape, and so fails checks on it.

Comment (KrisXV) [reply server/chat-plugins/chatlog.ts:1339]: You don't need to escape `-` in character classes unless it's between, a and z, A and Z, or 0 and 9.

Comment (Zarel) [reply server/chat-plugins/chatlog.ts:1339]: That's because it _is_ a useless escape.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**